Framework/PythonInterface/mantid/plots/helperfunctions.py

```diff
@@ -513,24 +513,28 @@ def get_axes_labels(workspace):
     :param workspace: :class:`mantid.api.MatrixWorkspace` or :class:`mantid.api.IMDHistoWorkspace`
     """
     if isinstance(workspace, MultipleExperimentInfos):
-        axes = ['Intensity']
+        axes_labels = ['Intensity']
         dims = workspace.getNonIntegratedDimensions()
         for d in dims:
             axis_title = d.name.replace('DeltaE', r'$\Delta E$')
             axis_unit = d.getUnits().replace('Angstrom^-1', r'$\AA^{-1}$')
             axis_unit = axis_unit.replace('DeltaE', 'meV')
             axis_unit = axis_unit.replace('Angstrom', r'$\AA$')
             axis_unit = axis_unit.replace('MomentumTransfer', r'$\AA^{-1}$')
-            axes.append('{0} ({1})'.format(axis_title, axis_unit))
+            axes_labels.append('{0} ({1})'.format(axis_title, axis_unit))
     else:
-        '''For matrix workspaces, return a tuple of ``(YUnit, <other units>)``'''
-        axes = [workspace.YUnit()]  # TODO: deal with distribution
+        # For matrix workspaces, return a tuple of ``(YUnit, <other units>)``
+        y_unit_label = workspace.YUnitLabel()
+        if ' per ' in y_unit_label:
```

Comment:
Thanks for fixing this `TODO` that had clearly been an oversight. I was wondering what you thought of defining this behaviour in the `MatrixWorkspace::YUnitLabel()` method instead by adding an argument called something like `useMathText` and then encapsulating all of this behaviour in 1 place? We can set it to `False` by default to keep the existing behaviour.